### PR TITLE
Wok 688

### PR DESCRIPTION
**Pull request by thomasegense**
Er der deployet på devel. Bemærk filterquery logges ved hver Solr search, så det er let at se hvad der sker.

Lige nu er der hardkodet "minimal" rettighed ved kald af licensemodulet.
(everybody=yes)
Og den er vigtig for rettigheder. Er der ingen attributer har man ingen rettigheder.

### CODE REVIEW

File: src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java [hunk 1]
```diff
@@ -167,12 +176,51 @@ public String solrSearch(String collection, String q, List<String> fq, Integer r
             SolrService solr = SolrManager.getSolrService(collection);
             // TODO: Pass the map of request parameters instead of all parameters as first class
             httpServletResponse.setContentType(solr.getResponseMIMEType(wt)); // Needed by SolrJ
+           
+             //Add filter query from license module.
+            DsLicenseApi licenseClient = getDsLicenseApiClient();
+            GetUserQueryInputDto licenseQueryDto = getLicenseQueryDto();
+            GetUsersFilterQueryOutputDto filterQuery = licenseClient.getUserLicenseQuery(licenseQueryDto);
+            log.info("filter query from licensemodule:"+filterQuery.getFilterQuery()); // in the start this is a very usefull log!
```
Comment (tokee) [inline src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java:184]: Please log the user attributes together with the resolved filter query for easier debugging.

File: src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java [hunk 2]
```diff
@@ -167,12 +176,51 @@ public String solrSearch(String collection, String q, List<String> fq, Integer r
...
 
+    
+    private static GetUserQueryInputDto getLicenseQueryDto() {
+    	   GetUserQueryInputDto getQueryDto = new GetUserQueryInputDto(); 
+          
+    	   getQueryDto.setPresentationType("Search"); // Important. Must be defined in Licensemodule with same name           
+         
+    	   //"everybody=true" is a value everyone will (from keycloak?)
+    	   UserObjAttributeDto everybodyUserAttribute=new UserObjAttributeDto();         
+           everybodyUserAttribute.setAttribute("everybody");
```
Comment (tokee) [inline src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java:201]: This is nice: Easy to change when we add OAuth2 capabilities.

File: src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java [hunk 3]
```diff
@@ -167,12 +176,51 @@ public String solrSearch(String collection, String q, List<String> fq, Integer r
...
+           
+           getQueryDto.setAttributes(allAttributes);           
+           return getQueryDto;
+    	
+    	
+    }
+    
+    private static DsLicenseApi getDsLicenseApiClient() {
+        
+        ApiClient client = Configuration.getDefaultApiClient();
```
Comment (tokee) [inline src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java:217]: Not used for anything as the `DsLicenseClient` takes care of it.

File: src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java [hunk 4]
```diff
@@ -167,12 +176,51 @@ public String solrSearch(String collection, String q, List<String> fq, Integer r
...
+           List<UserObjAttributeDto> allAttributes = new ArrayList<UserObjAttributeDto>(); 
+           allAttributes.add(everybodyUserAttribute);
+           
+           getQueryDto.setAttributes(allAttributes);           
+           return getQueryDto;
+    	
+    	
+    }
+    
+    private static DsLicenseApi getDsLicenseApiClient() {
```
Comment (tokee) [inline src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java:215]: A new DsLicenseClient will be created for every call to `solrSearch`. Could you make it a Singleton?

**Review by tokee — CHANGES_REQUESTED** [review]
A slight bit of coding work please.

File: src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java (hunk 4 above)
Comment (thomasegense) [reply src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java:215]: Done. Hoping it is thread safe :)

File: src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java (hunk 1 above)
Comment (thomasegense) [reply src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java:184]: very useful indeed for the endless debugging to come,

File: src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java (hunk 4 above)
Comment (tokee) [reply src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java:218]: It is. But thanks for the sanity check.

**Review by tokee — APPROVED** [review]

**Pull request merged**